bundles/org.eclipse.passage.lbc.server/src/org/eclipse/passage/lbc/server/jetty/Port.java

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2020 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lbc.server.jetty;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public final class Port implements Supplier<Integer> {
+
+	private final int port;
+
+	public Port(int port) {
+		this.port = port;
+	}
+
+	@Override
+	public Integer get() {
+		return port;
+	}
+
+	public static class OfArgument implements Function<String, Port> {
+
+		@Override
+		public Port apply(String argument) {
+			try {
+				return new Port(Integer.parseInt(argument));
+			} catch (NumberFormatException e) {
+				return new Port(8090);
```

Comment:
It's not a port-construction solution. 

It's better to throw an exception here and pull the fall-back logic to the upper level (it can be logged there or conveyed to a  client by other menas).  

Sticking to this particular design we need also 
```
public static final class Default implements Supplier<Port>
```
defined right here. 

BTW, in any places where 8090 is mentioned there should stay a code like `new Port.Default().get().get()`.

To ease a client code, you can define a `Port` itself as abstract with, and `OfArgument` and `Default` as final implementations.